src/context/todoProvider.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  ReactNode, createContext, useContext, useEffect, useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api/todos';
+import { Filter } from '../types/Filter';
+import { ErrorTypes } from '../types/errorTypes';
+import { USER_ID } from '../utils/userId';
+
+type TodosProps = {
+  todos: Todo[];
+  taskName: string;
+  setTaskName: (query: string) => void;
+  filterBy: Filter;
+  setFilterBy: (query: Filter) => void,
+  count: number;
+  error: null | string;
+  setError: (err: string | null) => void;
+};
+
+const TodosContext = createContext<TodosProps>({
+  todos: [],
+  taskName: '',
+  setTaskName: () => undefined,
+  filterBy: 'all',
+  setFilterBy: () => undefined,
+  count: 0,
+  error: null,
+  setError: () => undefined,
+});
+
+type Props = {
+  children: ReactNode;
+};
+
+const dataFilter = (data: Todo[], filtr: Filter) => {
+  switch (filtr) {
+    case 'active':
+      return data.filter(task => !task.completed);
+    case 'completed':
+      return data.filter(task => task.completed);
+    default:
+      return data;
+  }
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [taskName, setTaskName] = useState<string>('');
+  const [filterBy, setFilterBy] = useState<Filter>('all');
+  const [count, setCount] = useState<number>(0);
+  const [error, setError] = useState<null | string>(null);
+
+  const value = {
+    todos,
+    taskName,
+    setTaskName,
+    filterBy,
+    setFilterBy,
+    count,
+    error,
+    setError,
+  };
+
+  useEffect(() => {
+    try {
+      getTodos(USER_ID)
+        .then(data => setTodos(dataFilter(data, filterBy)));
+    } catch (err) {
+      setError(ErrorTypes.Load);
+    }
+
+    const counter = dataFilter(todos, 'active').length;
+
+    setCount(counter);
+  }, [todos, filterBy, count]);
```

Comment:
Open Network Tab in the devtools and see how many requests are being send.
It's because of dependencies array here. You definitely don't want `todos` to be part of it, because you update this array here, which leads to endless requests. Examine also the rest of dependencies.
Maybe you shouldn't fetch todos each time filter changes and just simply filter them?